lib/new_relic/environment_report.rb

```diff
@@ -71,7 +71,13 @@ def self.registered_reporters=(logic)
     report_on('OS'                ) { ::NewRelic::Agent::SystemInfo.ruby_os_identifier     }
     report_on('Database adapter'  ) do
       begin
-        ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, spec_name: "primary").config['adapter']
+        if ::ActiveRecord::Base.respond_to?(:connection_db_config)
+          ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, name: "primary")
```

Comment:
I tested out your fix with Rails 6.1 and I am actually seeing a NoMethodError arise from this line:

```
NoMethodError: undefined method `connection_db_config' for #<ActiveRecord::DatabaseConfigurations::HashConfig:0x00007fd54e72c6c8>
```

As that error implies, calling `connection_db_config` on `ActiveRecord::Base` directly will work:

```
::ActiveRecord::Base.connection_db_config.configuration_hash[:adapter]
```

But we really want to make sure we are accessing configuration for the agent's current environment, so instead we may want something more like this:

```
ActiveRecord::Base.configurations.configs_for(env_name: NewRelic::Control.instance.env, name: "primary").configuration_hash[:adapter]
```

And since we wouldn't be using `connection_db_config` in this case, it probably makes sense to change the conditional to be on Rails version, unless you can think of a better alternative.

Reply:
```
NoMethodError: undefined method `connection_db_config' for #<ActiveRecord::DatabaseConfigurations::HashConfig:0x00007fd54e72c6c8>
```

I did not get this error but I will try to recreate it.

> And since we wouldn't be using connection_db_config in this case, it probably makes sense to change the conditional to be on Rails version, unless you can think of a better alternative.

A condition on the Rails version sounds good to me.